### ISSUE TITLE
Support for other FFB devices too? 

### ISSUE THREAD

**Issue by SweetyVolty** [bobhelander/EDForceFeedback#4]
Hello, I have a saitek cyborg 3D FFB device. Is there any chance EDFF working with it too? 

Also lots of people using xbox controllers too which would be wonderful if it added too thanks ^^

Tried to change the guid with my devices but it never worked :'(

**Comment by bobhelander** [comment]
I put the product guid in the settings.json file as the very first setting.

It looks like this:
"ProductGuid": "001b045e-0000-0000-0000-504944564944",  = the Microsoft FFB2 Joystick

I also uploaded a new version that outputs the detected force feedback devices before the device is selected.

If your device shows up in the log, change the settings.json file with notepad and replace the product guid value with the id of your device.  Restart and see if it works.  

You can also try the fedit.exe utility under the "FFutils" folder and see if you can create effects for that device.

Please let me know if it does or does not.

**Comment by SweetyVolty** [comment]
Oh I wish my keyboard could able to talk, then it would tell you how many different GUID's I tried until my right hand literally gave up after a very long surfing on the web about what I was doing wrong while trying to compiling etc to adopt my devices to bring my tiny cockpit a little bit more action <3 I wanted to thank you even before trying so I will update very soon or ehm well that depends on my "talentlessness" heheh

o7

**Comment by SweetyVolty** [comment]
```
Event {
  "timestamp": "2020-04-01T01:37:33Z",
  "event": "ReservoirReplenished",
  "FuelMain": 16.749672,
  "FuelReservoir": 0.49
}
StatusEvent Status.Lights:True
Effect 13541c23-8e33-11d0-9ad0-00a0c9a06e35 starting
Effect Vibrate.ffe complete
StatusEvent Status.Lights:False
Effect 13541c23-8e33-11d0-9ad0-00a0c9a06e35 starting
Effect Vibrate.ffe complete
Event {
  "timestamp": "2020-04-01T01:39:07Z",
  "event": "SendText",
  "To": "local",
  "Message": "ehehehehe",
  "Sent": true
}
StatusEvent Status.Lights:True
Effect 13541c23-8e33-11d0-9ad0-00a0c9a06e35 starting
Effect Vibrate.ffe complete
StatusEvent Status.Lights:False
Effect 13541c23-8e33-11d0-9ad0-00a0c9a06e35 starting
Effect Vibrate.ffe complete``
```


I think the local sent text explains a lot :3

**Comment by bobhelander** [comment]
May I ask you to try these steps.  Download the latest release https://github.com/bobhelander/EDForceFeedback/releases/latest.  Make sure the joystick is plugged in and turned on then just run the EDForceFeedBack.exe once and reply with the output.  The device detection happens at the beginning and there is no need to run the program too long.

The output should look something like this:

Info    ─  Starting EliteAPI.
Info    ─  A new update (2.3.0.0) is available. Visit github.com/EliteAPI/EliteAPI to download the latest version.
Info    ─  Journal directory set to 'C:\Users\Bob\Saved Games\Frontier Developments\Elite Dangerous'.
Info    ─  Found Journal and Status files.
Warning ─  Elite: Dangerous is not in-game.
Warning ┬  Could not find event class 'SquadronStartupInfo'.
        ├  Sequence contains no elements
        ╞  at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
        ╞  at EliteAPI.JournalParser.ProcessJson(String json)
Warning ┬  Could not find event class 'SquadronStartupInfo'.
        ├  Sequence contains no elements
        ╞  at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
        ╞  at EliteAPI.JournalParser.ProcessJson(String json)
Success ─  EliteAPI is ready.
Info    ─  Starting rich presence.
No joystick/Gamepad found.
Success ─  Discord Rich Presence has connected and is running.

Thanks,
Bob

**Comment by SweetyVolty** [comment]
tried with the last; 

Xbox One controller Bluetooth Wireless; 

Info    ─  Starting EliteAPI.
Info    ─  A new update (2.3.0.0) is available. Visit github.com/EliteAPI/EliteAPI to download the latest version.
Info    ─  Journal directory set to 'C:\Users\illum\Saved Games\Frontier Developments\Elite Dangerous'.
Info    ─  Found Journal and Status files.
Success ─  EliteAPI is ready.
Info    ─  Starting rich presence.
No joystick/Gamepad found.
Success ─  Discord Rich Presence has connected and is running.
 

Xbox one controller with wired gives same output. 


With Saitek it worked without GUID input at settings;  

Info    ─  Starting EliteAPI.
Info    ─  A new update (2.3.0.0) is available. Visit github.com/EliteAPI/EliteAPI to download the latest version.
Info    ─  Journal directory set to 'C:\Users\illum\Saved Games\Frontier Developments\Elite Dangerous'.
Info    ─  Found Journal and Status files.
Success ─  EliteAPI is ready.
Info    ─  Starting rich presence.
Device: Cyborg 3D Force Stick: ProductGuid ff1206a3-0000-0000-0000-504944564944
No joystick/Gamepad found.
Success ─  Discord Rich Presence has connected and is running.

With GUID; 

Info    ─  Starting EliteAPI.
Info    ─  A new update (2.3.0.0) is available. Visit github.com/EliteAPI/EliteAPI to download the latest version.
Info    ─  Journal directory set to 'C:\Users\illum\Saved Games\Frontier Developments\Elite Dangerous'.
Info    ─  Found Journal and Status files.
Success ─  EliteAPI is ready.
Info    ─  Starting rich presence.
Device: Cyborg 3D Force Stick: ProductGuid ff1206a3-0000-0000-0000-504944564944
Found Joystick/Gamepad with GUID: b2bf6990-3de7-11ea-8002-444553540000
Effect available Constant Force
Effect available Ramp Force
Effect available Square Wave
Effect available Sine Wave
Effect available Triangle Wave
Effect available SawtoothUp Wave
Effect available Sawtooth Down Wave
Effect available Spring Force
Effect available Damper Force
Effect available Inertia
Effect available Friction
Effect available Custom Force
Effect available Axis Barrier
Effect available Angle Barrier
Effect available Axis Wall
Effect available Angle Wall
File Effect available Cargo.ffe
File Effect available Dock.ffe
File Effect available Gear.ffe
File Effect available Hardpoints.ffe
File Effect available Landed.ffe
File Effect available Vibrate.ffe
File Effect available VibrateSide.ffe
Success ─  Discord Rich Presence has connected and is running.

**Comment by SweetyVolty** [comment]
Oh and both connected this happens but when I insert the xbox's guid nothing happens. BTW, I installed a xbox driver with all of the same output as saitek so it should've been work as it works with direct input tools. 

Info    ─  Starting EliteAPI.
Info    ─  A new update (2.3.0.0) is available. Visit github.com/EliteAPI/EliteAPI to download the latest version.
Info    ─  Journal directory set to 'C:\Users\illum\Saved Games\Frontier Developments\Elite Dangerous'.
Info    ─  Found Journal and Status files.
Success ─  EliteAPI is ready.
Info    ─  Starting rich presence.
Device: Cyborg 3D Force Stick: ProductGuid ff1206a3-0000-0000-0000-504944564944
Found Joystick/Gamepad with GUID: b2bf6990-3de7-11ea-8002-444553540000
Effect available Constant Force
Effect available Ramp Force
Effect available Square Wave
Effect available Sine Wave
Effect available Triangle Wave
Effect available SawtoothUp Wave
Effect available Sawtooth Down Wave
Effect available Spring Force
Effect available Damper Force
Effect available Inertia
Effect available Friction
Effect available Custom Force
Effect available Axis Barrier
Effect available Angle Barrier
Effect available Axis Wall
Effect available Angle Wall
File Effect available Cargo.ffe
File Effect available Dock.ffe
File Effect available Gear.ffe
File Effect available Hardpoints.ffe
File Effect available Landed.ffe
File Effect available Vibrate.ffe
File Effect available VibrateSide.ffe
Success ─  Discord Rich Presence has connected and is running.

**Comment by SweetyVolty** [comment]
Sorry I don't know if I'm bothering you but do you have a discord? Azorath#7017 <3 

Also dou you have a list of effects list from the eliteapi not able to pull from them yet.

**Comment by bobhelander** [comment]
It looks as if the Cyborg should work.  The Xbox controller is just a vibrate and I believe it already is supported in Elite directly.  I have sent the Discord request (from user Axe) and we should be able to get the Cyborg working.

**Comment by bobhelander** [comment]
Thank you for the help.  Your suggestions have been implemented and the new version 2.0.0.0 is available.  For future travelers to this thread, the xbox driver information can be found here: https://steamcommunity.com/discussions/forum/1/541907867790900215/  and the driver is archived here: https://onedrive.live.com/?authkey=%21AFA%2D6FlVglum%2DJI&id=4E84FBD1E7FE851%2118634&cid=04E84FBD1E7FE851
